diff -puNr gnome-vfs-2.19.2.orig/libgnomevfs/gnome-vfs-ace.c gnome-vfs-2.19.2/libgnomevfs/gnome-vfs-ace.c
--- gnome-vfs-2.19.2.orig/libgnomevfs/gnome-vfs-ace.c	2007-04-02 16:16:48.000000000 +0200
+++ gnome-vfs-2.19.2/libgnomevfs/gnome-vfs-ace.c	2007-08-28 13:13:07.000000000 +0200
@@ -119,70 +119,6 @@ gnome_vfs_ace_class_init (GnomeVFSACECla
 	gobject_class->finalize     = gnome_vfs_ace_finalize;
 	gobject_class->set_property = gnome_vfs_ace_set_property;
 	gobject_class->get_property = gnome_vfs_ace_get_property;
-	
-	/* properties */
-	
-	pspec =  g_param_spec_uint ("kind",
-				    "fixme",
-				    "fixme",
-				    GNOME_VFS_ACL_KIND_NULL,
-				    G_MAXUINT32,
-				    GNOME_VFS_ACL_KIND_NULL,
-				    G_PARAM_READWRITE |
-				    G_PARAM_CONSTRUCT);
-
-	g_object_class_install_property (gobject_class,
-					 PROP_KIND,
-					 pspec);
-
-
-
-	pspec =  g_param_spec_string ("id",
-				      "fixme",
-				      "fixme",
-				      NULL,
-				      G_PARAM_READWRITE |
-				      G_PARAM_CONSTRUCT);
-
-	g_object_class_install_property (gobject_class,
-					 PROP_ID,
-					 pspec);
-
-
-
-	pspec = g_param_spec_pointer ("permissions",
-				      "Permissions",
-				      "fixme",
-				      G_PARAM_READWRITE |
-				      G_PARAM_CONSTRUCT);
-
-	g_object_class_install_property (gobject_class,
-					 PROP_PERMS,
-					 pspec);
-
-
-
-	pspec = g_param_spec_boolean ("negative",
-				      "fixme",
-				      "fixme",
-				      FALSE,
-				      G_PARAM_READWRITE);
-
-	g_object_class_install_property (gobject_class,
-					 PROP_NEGATIVE,
-					 pspec);
-
-
-	pspec = g_param_spec_boolean ("inherit",
-				      "fixme",
-				      "fixme",
-				      FALSE,
-				      G_PARAM_READWRITE);
-
-	g_object_class_install_property (gobject_class,
-					 PROP_INHERIT,
-					 pspec);
-	
 }
 
 static void
Binary files gnome-vfs-2.19.2.orig/libgnomevfs/.Makefile.am.swp and gnome-vfs-2.19.2/libgnomevfs/.Makefile.am.swp differ
