--- vte-0.15.3/src/vte.h.orig	2007-02-17 19:38:45.000000000 +0100
+++ vte-0.15.3/src/vte.h	2007-02-18 09:23:05.000000000 +0100
@@ -330,27 +330,19 @@
  * VteCharAttributes structure in the passed GArray, if the array was not NULL.
  * Note that it will have one entry per byte, not per character, so indexes
  * should match up exactly. */
+typedef gboolean (*VteIsSelectedFunc) (VteTerminal *terminal, glong column, glong row, gpointer data);
 char *vte_terminal_get_text(VteTerminal *terminal,
-			    gboolean(*is_selected)(VteTerminal *terminal,
-						   glong column,
-						   glong row,
-						   gpointer data),
+			    VteIsSelectedFunc is_selected,
 			    gpointer data,
 			    GArray *attributes);
 char *vte_terminal_get_text_include_trailing_spaces(VteTerminal *terminal,
-						    gboolean(*is_selected)(VteTerminal *terminal,
-									   glong column,
-									   glong row,
-									   gpointer data),
+						    VteIsSelectedFunc is_selected,
 						    gpointer data,
 						    GArray *attributes);
 char *vte_terminal_get_text_range(VteTerminal *terminal,
 				  glong start_row, glong start_col,
 				  glong end_row, glong end_col,
-				  gboolean(*is_selected)(VteTerminal *terminal,
-							 glong column,
-							 glong row,
-							 gpointer data),
+				  VteIsSelectedFunc is_selected,
 				  gpointer data,
 				  GArray *attributes);
 void vte_terminal_get_cursor_position(VteTerminal *terminal,
