diff -puNr GConf-2.19.1.orig/gconf/gconf-changeset.h GConf-2.19.1/gconf/gconf-changeset.h
--- GConf-2.19.1.orig/gconf/gconf-changeset.h	2007-01-07 19:34:00.000000000 +0100
+++ GConf-2.19.1/gconf/gconf-changeset.h	2007-08-28 10:18:38.000000000 +0200
@@ -43,10 +43,6 @@ typedef void (* GConfChangeSetForeachFun
 
 gboolean        gconf_engine_commit_change_set   (GConfEngine* conf,
                                                   GConfChangeSet* cs,
-                                                  /* remove all
-                                                     successfully
-                                                     committed changes
-                                                     from the set */
                                                   gboolean remove_committed,
                                                   GError** err);
 
diff -puNr GConf-2.19.1.orig/gconf/gconf-client.h GConf-2.19.1/gconf/gconf-client.h
--- GConf-2.19.1.orig/gconf/gconf-client.h	2007-01-07 19:34:00.000000000 +0100
+++ GConf-2.19.1/gconf/gconf-client.h	2007-08-28 10:18:17.000000000 +0200
@@ -361,10 +361,6 @@ void         gconf_client_value_changed 
 
 gboolean        gconf_client_commit_change_set   (GConfClient* client,
                                                   GConfChangeSet* cs,
-                                                  /* remove all
-                                                     successfully
-                                                     committed changes
-                                                     from the set */
                                                   gboolean remove_committed,
                                                   GError** err);
 
